File: MajorityElement.scala
object Solution {
    def majorityElement(nums: Array[Int]): Int = {
        var count = 1;
        var maj = nums(0);
        for (i <- 1 until nums.length) {
            if (count == 0) {
                maj = nums(i);
                count += 1;
            } else if (maj == nums(i)) {
                count += 1;
            } else {
                count -= 1;
            }
        }
        maj;
    }
}

File: Array/MajorityElement.java
class Solution {
    public int majorityElement(int[] nums) {
        int maj = nums[0];
        int max = 1;
        for (int i = 1; i < nums.length; i++) {
            if(nums[i] == maj) {
                max++;
            } else if (max == 0) {
                maj = nums[i];
            } else {
                max--;
            }
        }
        return maj;

    }
}
